modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -409,7 +413,15 @@ private void onUpdatePartitionDistributionResetBusy(
             }
 
             if (partitionDistributionResetTimeoutSeconds == IMMEDIATE_TIMER_VALUE) {
-                // TODO: IGNITE-23599 Implement valid behaviour here.
+                fireEvent(
```

Comment:
I thought that we've agreed that we won't call event in MS Watch thread, aren't we?

Reply:
Fixed